File: movil/proyecto_integrado_flutter/muevet_flutter/lib/models/vehicle.dart
class Vehicle {
  final String licensePlate;
  final String brand;
  final String model;
  final int year;
  final double rentalPricePerDay;
  final bool available;
  final String type;

  Vehicle({
    required this.licensePlate,
    required this.brand,
    required this.model,
    required this.year,
    required this.rentalPricePerDay,
    required this.available,
    required this.type,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      licensePlate: json['licensePlate'],
      brand: json['brand'],
      model: json['model'],
      year: json['year'],
      rentalPricePerDay: (json['rentalPricePerDay'] as num).toDouble(),
      available: json['available'],
      type: json['type'], 
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'licensePlate': licensePlate,
      'brand': brand,
      'model': model,
      'year': year,
      'rentalPricePerDay': rentalPricePerDay,
      'available': available,
      'type': type,
    };
  }
}

File: backend/src/main/java/com/example/demo/entity/Vehicle.java
package com.example.demo.entity;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;

@Entity
public class Vehicle {
    @Id
    @Column(nullable = false, unique = true)
    private String licensePlate;


     @Column(nullable = false)
    private String brand;

    @Column(nullable = false)
    private String model;

    @Column(nullable = false)
    private int year;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal rentalPricePerDay;

    @Column(nullable = false)
    private boolean available = true;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private VehicleType type;

    public Vehicle() {}

    public Vehicle(String brand, String model, int year, BigDecimal rentalPricePerDay, String licensePlate, VehicleType type) {
        this.brand = brand;
        this.model = model;
        this.year = year;
        this.rentalPricePerDay = rentalPricePerDay;
        this.available = true;
        this.licensePlate = licensePlate;
        this.type = type;
    }

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public BigDecimal getRentalPricePerDay() {
        return rentalPricePerDay;
    }

    public void setRentalPricePerDay(BigDecimal rentalPricePerDay) {
        this.rentalPricePerDay = rentalPricePerDay;
    }

    public boolean isAvailable() {
        return available;
    }

    public void setAvailable(boolean available) {
        this.available = available;
    }

    
    
    public VehicleType getType() {
        return type;
    }

    public void setType(VehicleType type) {
        this.type = type;
    }

    public String getLicensePlate() {
        return licensePlate;
    }

    public void setLicensePlate(String licensePlate) {
        this.licensePlate = licensePlate;
    }

    
}
